test/libsolidity/semanticTests/inlineAssembly/transient_storage_low_level_calls.sol

```diff
@@ -68,3 +68,9 @@ contract C {
 // testCall() -> true
 // tloadAllowedStaticCall() -> true
 // tstoreNotAllowedStaticCall() -> true
+// gas irOptimized: 98419720
+// gas irOptimized code: 19000
+// gas legacy: 98409086
+// gas legacy code: 30000
+// gas legacyOptimized: 98420962
+// gas legacyOptimized code: 17800
```

Comment:
I hadn't noticed that. I will try to check where exactly it is getting that high cost.

Reply:
ok. This looks highly suspicious so it would be great if we knew what's happening and if something is broken.

But that's just in case. This is not a blocker for the PR itself. I think it's unlikely that it's caused by it and it's probably that we're just seeing the effects of something else here.

Reply:
True! And 100000000 is the gas limit in our tests: https://github.com/ethereum/solidity/blob/cc79c91b93d322cd38beec08a1ce575d3622c7e4/test/ExecutionFramework.h#L275

So mystery solved. Looks like a failed `staticcall` just eats all gas forwarded to it. I did not consider that this might happen in cases other than hitting an invalid opcode. Or is it simply because state-mutating opcodes are considered invalid within a `staticcall`?

Reply:
Yeah, it just hits an invalid opcode and thus eats up all gas forwarded to it.